opm/simulators/flow/MixingRateControls.cpp

```diff
@@ -137,22 +137,49 @@ drsdtConvective(int episodeIdx) const
     return (oilVaporizationControl.drsdtConvective());
 }
 
+template<class FluidSystem>
+bool MixingRateControls<FluidSystem>::
+drsdtActive(int episodeIdx, std::size_t pvtRegionIdx) const
+{
+    const auto& oilVaporizationControl = schedule_[episodeIdx].oilvap();
+    return (oilVaporizationControl.drsdtActive(pvtRegionIdx));
+}
+
+template<class FluidSystem>
+bool MixingRateControls<FluidSystem>::
+drvdtActive(int episodeIdx, std::size_t pvtRegionIdx) const
+{
+    const auto& oilVaporizationControl = schedule_[episodeIdx].oilvap();
+    return (oilVaporizationControl.drvdtActive(pvtRegionIdx));
+}
+
+template<class FluidSystem>
+bool MixingRateControls<FluidSystem>::
+drsdtConvective(int episodeIdx, std::size_t pvtRegionIdx) const
+{
+    const auto& oilVaporizationControl = schedule_[episodeIdx].oilvap();
+    return (oilVaporizationControl.drsdtConvective(pvtRegionIdx));
+}
+
+
 template<class FluidSystem>
 void MixingRateControls<FluidSystem>::
 updateExplicitQuantities(const int episodeIdx,
                          const Scalar timeStepSize)
 {
     const auto& oilVaporizationControl = schedule_[episodeIdx].oilvap();
-    if (this->drsdtActive(episodeIdx)) {
-        // DRSDT is enabled
-        for (std::size_t pvtRegionIdx = 0; pvtRegionIdx < maxDRs_.size(); ++pvtRegionIdx)
+    // DRSDT is enabled
```

Comment:
Here, we already have access to `oilVaporizationControl`, then `oilVaporizationControl.drsdtActive(pvtRegionIdx)` will be able to determine whether it is active. In my opinion, we do not need to go through the function `drvdtActive(episodeIdx, pvtRegionIdx)` .  Basically, it will create another reference to `oilVaporizationControl`.

Reply:
Good point. I will update.